uniss.js

```diff
@@ -0,0 +1,107 @@
+const options = {
+  prefix: "un", // class prefix
+  inline: false, // set css as inline, applies only to ones without :state and @media
+  internal: true, // set classes in internall <style>
+  mediaQueries: [
+    {
+      name: "sm",
+      value: "@media (min-width: 640px)",
+    },
+    {
+      name: "md",
+      value: "@media (min-width: 768px)",
+    },
+    {
+      name: "lg",
+      value: "@media (min-width: 1024px)",
+    },
+    {
+      name: "xl",
+      value: " @media (min-width: 1280px)",
+    },
+  ],
+  pseudoStates: ["hover", "active", "visited", "focus", "focus-within"],
+};
+
+window.onload = function () {
+  const ourStyleElement = document.createElement("style");
+  if (options.internal) {
+    // if enabled adds <style> with our css to head
+    document.head.appendChild(ourStyleElement);
+  }
+
+  const searchQuery = `[class*='${options.prefix}-']`; // search string for elemnt selection
```

Comment:
we should definitely not check and do querySelector based on the `un` prefix.

to save bandwidth, a better approach would be to generate all css at runtime and inject it into the document. Doing this only for elements that match query selector, would mean elements inserted later would not be matched

Reply:
but first it's worth knowing the size of the whole css, after minification and gzip is applied.

Reply:
> we should definitely not check and do querySelector based on the un prefix.

we shouldn't use querySelector at all or ?
we just get elements with prefix in className and then filter those classes to make sure is valid 

> to save bandwidth, a better approach would be to generate all css at runtime and inject it into the document. Doing this only for elements that match query selector, would mean elements inserted later would not be matched

Static properties like display, position.. we can generate as static CSS file, and other properties like width, height... we can generate on runtime, at page load and at each change of class.

We can use [MutationObserver ](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) for looking at those changes